### PR TITLE
Patched !welcome script (default name and API pagination)

### PR DESCRIPTION
**Pull request by mitchmcdee**
As per title, fixes two bugs from #166

1) display_name is optional, so instead we now default to `display_name` and resort to `name` as a back up (which is a required field)
2) Slack Web API default limits `conversations.members` to returning only 100 members. Added pagination code that utilises `next_cursor` to accurately yield _all_ members

### CODE REVIEW

**Comment by mitchmcdee** [comment]
I just realised this code "fixes" that it will only fire when a user enters #announcements instead of #general. Should I revert it back to firing on #general?

**Comment by ashfordneil** [comment]
The getNumMembers function should probably use the `user` argument that was passed to it rather than relying on `announcements` being defined in a previous scope. Also is there any chance that the slack api offers a way for us to get the number of members in one go without iterating through them one by one? 

Other than those things, assuming you’ve tested it, it looks good to me. (Unfortunately I can’t test it myself at the moment)

**Comment by mitchmcdee** [comment]
Interesting, any reason why you'd argue to use `user` over `announcements`? (or did you make a mistake, because user isn't passed to getNumMembers?)

re: slack API, unfortunately it seems like no-- not that I've seen at least. The only way seems to be getting a members list, of which only pagination-type methods exist as per later this year (https://api.slack.com/changelog/2017-10-members-array-truncating). On that note though, It seems I can simplify the code because `conversations.members` doesn't seem to include deleted/deactivated users, something we had to take into account when using `users.list`

Fair about testing, it's a bit of a pain to setup anyway bahahhaa

File: scripts/welcome.js [hunk 1]
```diff
@@ -13,44 +14,48 @@ var WELCOME_MESSAGES = [
...
-        });
+        // Get member count to see if we've hit a member milestone
+        getMemberCount(robot, announcements, 0).then(memberCount => {
+            // If we're not at a member milestone, don't bother celebrating!
+            if (memberCount % MEMBER_MILESTONE != 0) {
+                return;
+            }
+
+            res.send(":tada: " + memberCount + " members! :tada:");
+        }).catch(err => console.log(err));
```
Comment (CameronAavik) [inline scripts/welcome.js]: can eta-reduce this to `.catch(console.log)` :P

Comment (mitchmcdee) [reply scripts/welcome.js]: good catch :^) done

**Comment by mitchmcdee** [comment]
I'm going to wait for Neil's response/approval before merging :^)

**Pull request merged**